hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/AbstractGenericRecordBuilder.java

```diff
@@ -129,6 +133,14 @@ public GenericRecordBuilder setTimestampWithTimezone(@Nonnull String fieldName,
     @Nonnull
     @Override
     public GenericRecordBuilder setArrayOfGenericRecord(@Nonnull String fieldName, @Nullable GenericRecord[] value) {
+        if (value != null) {
+            for (GenericRecord genericRecord : value) {
+                if (!(genericRecord instanceof CompactGenericRecord)) {
```

Comment:
this will not work when the `genericRecord` is null, as the instanceof check will return false, but I believe it is OK to have it null

Reply:
added genericRecord != null check